ui-tests/src/test/java/io/jenkins/plugins/analysis/warnings/UiTest.java

```diff
@@ -251,6 +254,43 @@ protected void verifyCheckStyle(final Build build) {
                 .hasErrorMessages("Can't create fingerprints for some files:");
     }
 
+    protected void verifyPep8(final Build build) {
+        verifyPep8(build, 1);
+    }
+
+    protected void verifyPep8(final Build build, final int referenceBuild) {
+        build.open();
+        AnalysisSummary pep8 = new AnalysisSummary(build, PEP8_ID);
+        assertThat(pep8).isDisplayed()
+                .hasTitleText(PEP8_NAME + ": 8 warnings")
+                .hasReferenceBuild(referenceBuild)
+                .hasInfoType(InfoType.ERROR);
+
+        AnalysisResult pep8Details = verifyPep8Details(pep8);
+
+        pep8Details.openTab(Tab.ISSUES);
+        IssuesDetailsTable issuesTable = pep8Details.openIssuesTable();
+        assertThat(issuesTable).hasSize(8);
+
+        long normalIssueCount = issuesTable.getTableRows().stream()
+                .map(row -> row.getAs(IssuesTableRow.class).getSeverity())
+                .filter(severity -> severity.equals("Normal")).count();
+
+        long lowIssueCount = issuesTable.getTableRows().stream()
+                .map(row -> row.getAs(IssuesTableRow.class).getSeverity())
+                .filter(severity -> severity.equals("Low")).count();
+
+        assertThat(normalIssueCount).isEqualTo(6);
+        assertThat(lowIssueCount).isEqualTo(2);
+    }
+
+    protected AnalysisResult verifyPep8Details (final AnalysisSummary pep8) {
+        AnalysisResult pep8Details = pep8.openOverallResult();
+        assertThat(pep8Details).hasActiveTab(Tab.ISSUES)
```

Comment:
In den beiden Smoke Tests ist beim Öffnen der Details der Issues-Tab aktiv (und nicht der Categories-Tab). Daher musste ich auch beim GlobalConfigurationUiTest die Methode wieder überschreiben, damits wieder passt.

Ist das normal, dass an dieser Stelle der aktive Tab auf einmal anders ist?
Weil jedes andere Plugin und der Groovy Parser selbst (wenn er das einzige Plugin ist) hier eben den Categories-Tab aktiv hat.

Reply:
Der aktive Tab wird gespeichert (im Browser local storage). So hat man als Nutzer immer den letzten aktiven Tab sichtbar, egal wo man hinnavigiert. In den Tests hat sich das unbewusst so ergeben, da die ersten Tests nur den Issues Tab anzeigen.

Reply:
Ahh cool, ich wusste gar nicht, dass an der Stelle sogar gecached wird. Dann passt das Verhalten ja :)